src/databricks/labs/ucx/account.py

```diff
@@ -183,13 +190,24 @@ def _get_account_groups(self) -> dict[str | None, list[ComplexValue] | None]:
         for acc_grp_id in self._ac.groups.list(attributes="id"):
             if not acc_grp_id.id:
                 continue
-            full_account_group = self._ac.groups.get(acc_grp_id.id)
-            logger.debug(f"Found account group {acc_grp_id.display_name}")
+            full_account_group = self._safe_groups_get(self._ac, acc_grp_id.id)
+            if not full_account_group:
+                continue
+            logger.debug(f"Found account group {full_account_group.display_name}")
             acc_groups[full_account_group.display_name] = full_account_group.members
 
         logger.info(f"{len(acc_groups)} account groups found")
         return acc_groups
 
+    def _safe_groups_get(self, interface, group_id) -> Group | None:
```

Comment:
technically the issue is that Identity API is eventually consistent, so we need to retry with exponential backoff, and the upper limit should be something reasonable like 2m

Reply:
This feature is not a heavy create/get workload where indeed it would make sense to retry in this case.

The problem is that the code is doing a list, then iterates over the result of that list to get the group details. But it can happen that the group has been deleted after it was captured by the list command, then the get would fail.

This scenario happens because in the case of our integration tests, we create/delete a huge amount of groups in parallel.